app/Http/Controllers/AdminController.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Hash;
+use App\User;
+use Validator;
+use Illuminate\Http\Request;
+use App\Http\Resources\Admin as AdminResource;
+
+class AdminController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $admins = User::whichAdmin()->paginate(5);
```

Comment:
suggestion for the pagination:
Add a constant PER_PAGE = 5 in parent Controller class.
Then you'll use it in other controllers also.